modules/ui_prefs.py

```
@@ -0,0 +1,207 @@
+        }
+
+        directory = os.path.dirname(self._prefs_path) or '.'
+        tmp_path = self._prefs_path + '.tmp'
+
+        try:
+            os.makedirs(directory, exist_ok=True)
+            with open(tmp_path, 'w', encoding='utf-8') as f:
+                json.dump(payload, f, indent=2)
+            os.replace(tmp_path, self._prefs_path)
+        except OSError as e:
+            logger.warning(f"Failed to save UI prefs to {self._prefs_path}: {e}")
```

Ruff (0.16.2)

[warning] Logging statement uses f-string

(G004)